### PR TITLE
Spellcheck.js: Improve language detection

### PR DESCRIPTION
**Pull request by rhythm-sharma**
---

**What's this PR do?**

**Any background context you want to provide?**

**Screenshots?**

**You have tested this PR on:**
  - [ ] Windows
  - [ ] Linux/Ubuntu
  - [ ] macOS

### CODE REVIEW

File: app/renderer/js/preload.js [hunk 1]
```diff
@@ -45,13 +45,13 @@ process.once('loaded', () => {
...
-			// Init spellchecker
-			SetupSpellChecker.init();
-		}
+		// const serverLanguage = page_params.default_language; // eslint-disable-line no-undef, camelcase
+		// if (serverLanguage) {
+		// 	// Set spellcheker language
+		// 	ConfigUtil.setConfigItem('spellcheckerLanguage', serverLanguage);
+		// 	// Init spellchecker
+		//SetupSpellChecker.init();
+		// }
```
Comment (kanishk98) [inline app/renderer/js/preload.js:54]: Please remove the commented-out part. Don't see why this has to stay in.

File: app/renderer/js/spellchecker.js [hunk 2]
```diff
@@ -2,6 +2,12 @@
 
 const { SpellCheckHandler, ContextMenuListener, ContextMenuBuilder } = require('electron-spellchecker');
 
+const LanguageDetect = require('languagedetect');
+
+const lngDetector = new LanguageDetect();
+
+const ISO6391 = require('iso-639-1');
```
Comment (kanishk98) [inline app/renderer/js/spellchecker.js:9]: If you've added new `npm` modules here, please also add those to your `package.json` and push the changes.

File: app/renderer/js/spellchecker.js [hunk 3]
```diff
@@ -26,13 +32,12 @@ class SetupSpellChecker {
...
 		if (this.SpellCheckHandler) {
-			this.SpellCheckHandler.attachToInput();
-
-			const userLanguage = ConfigUtil.getConfigItem('spellcheckerLanguage');
-
-			this.SpellCheckHandler.switchLanguage(userLanguage);
+			setTimeout(() => this.SpellCheckHandler.attachToInput(), 1000);
+			const langname = lngDetector.detect(input, 1);
+			const langcode = ISO6391.getCode(langname);
+			this.SpellCheckHandler.switchLanguage(langcode);
```
Comment (kanishk98) [inline app/renderer/js/spellchecker.js:40]: I'm guessing that this is meant to detect the language the user is typing and then change the spellchecker's language to that, but I don't see why the additional modules are required if `electron-spellchecker` is enough for that. 

(that's what I think this PR is meant for, please correct me if I'm wrong).

**Review by kanishk98 — COMMENTED** [review]
@rrhythmsharma submitted a prelim review. It'd be great if you could update the description for the PR with the intended changes, any screenshots, and the reasons for those extra modules. 

Thanks!

**Comment by akashnimare** [comment]
Closing this in favour of #752.

**Pull request closed without merging**